src/Blueprint.php

```diff
@@ -82,42 +85,37 @@ public function expectToOnlyUse(LayerOptions $options, callable $failure): void
                     // @phpstan-ignore-next-line
                     fn (ObjectDescription $object): string => $object->name, iterator_to_array($layer->getIterator())), array_map(
                         fn (string $dependency): Layer => $this->layerFactory->make($options, $dependency),
-                        [$targetValue, ...array_map(
-                            fn (Dependency $dependency): string => $dependency->value, $this->dependencies->values
-                        )],
+                        [
+                            $targetValue, ...array_map(
+                                fn (Dependency $dependency): string => $dependency->value, $this->dependencies->values
+                            ),
+                        ],
                     )
                 ));
 
-            $notDeclaredDependencies = [];
-
-            foreach ($this->layerFactory->make($options, $targetValue) as $object) {
+            $layer = $this->layerFactory->make($options, $targetValue);
+            foreach ($layer as $object) {
                 assert($object instanceof ObjectDescription);
 
                 foreach ($object->uses as $use) {
                     assert(is_string($use));
 
                     if (! in_array($use, $allowedUses, true)) {
-                        $notDeclaredDependencies[] = $use;
```

Comment:
@nunomaduro I refactored this one to immediately trigger the `$failure` closure, as in line 107 we always return the first occurrence found